lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -214,4 +232,120 @@ void LLDBMemoryReader::popLocalBuffer() {
   m_local_buffer_size = 0;
 }
 
+llvm::Optional<std::pair<uint64_t, uint64_t>>
+LLDBMemoryReader::addModuleToAddressMap(ModuleSP module) {
+  // The first available address is the mask, since subsequent images are mapped
+  // in ascending order, all of them will contain this mask.
+  uint64_t module_start_address = TAGGED_ADDRESS_MASK;
+  if (!m_range_module_map.empty())
+    // We map the images contiguously one after the other, all with the tag bit
+    // set.
+    // The address that maps the last module is exactly the address the new
+    // module should start at.
+    module_start_address = m_range_module_map.back().first;
+
+#ifndef NDEBUG
+  static std::initializer_list<uint64_t> used_swift_masks = {
+      SWIFT_ABI_ARM_OBJC_WEAK_REFERENCE_MARKER_MASK,
+      SWIFT_ABI_ARM_OBJC_WEAK_REFERENCE_MARKER_VALUE,
+      SWIFT_ABI_ARM_IS_OBJC_BIT,
+      SWIFT_ABI_X86_64_SWIFT_SPARE_BITS_MASK,
+      SWIFT_ABI_X86_64_OBJC_RESERVED_BITS_MASK,
+      SWIFT_ABI_X86_64_SIMULATOR_OBJC_RESERVED_BITS_MASK,
+      SWIFT_ABI_X86_64_IS_OBJC_BIT,
+      SWIFT_ABI_X86_64_OBJC_WEAK_REFERENCE_MARKER_MASK,
+      SWIFT_ABI_X86_64_SIMULATOR_OBJC_WEAK_REFERENCE_MARKER_MASK,
+      SWIFT_ABI_X86_64_SIMULATOR_OBJC_WEAK_REFERENCE_MARKER_VALUE,
+      SWIFT_ABI_ARM64_SWIFT_SPARE_BITS_MASK,
+      SWIFT_ABI_ARM64_OBJC_RESERVED_BITS_MASK,
+      SWIFT_ABI_ARM64_IS_OBJC_BIT,
+      SWIFT_ABI_ARM64_OBJC_WEAK_REFERENCE_MARKER_MASK};
+
+  for (auto swift_mask : used_swift_masks) {
+    assert((module_start_address & swift_mask) != swift_mask &&
+           "LLDB tagged address clashes with a swift mask!");
```

Comment:
This is now checking that the start address doesn't use any masked bits. That's good I think we also should check that our own tag bit don't conflict with any runtime bits.

Reply:
What do you mean by our own tag bit? If `TAGGED_ADDRESS_MASK` itself conflicts with runtime bits?